sycl/source/detail/non_cached_kernel_lock.cpp

```diff
@@ -0,0 +1,22 @@
+#include "non_cached_kernel_lock.hpp"
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace detail {
+
+NonCachedKernelLock::~NonCachedKernelLock() {}
+
+Locked<RT::PiKernel> NonCachedKernelLock::lockKernel(RT::PiKernel &K) {
+  std::mutex &Mtx = Map[K];
```

Comment:
Shouldn't access to the `Map` itself be under a lock?

Reply:
Right.
This is a major bug here...

Reply:
Fixed.